subsys/bluetooth/audio/has_client.c

```diff
@@ -82,11 +88,176 @@ static enum bt_has_capabilities get_capabilities(const struct has_inst *inst)
 	return caps;
 }
 
+union preset_record {
+	struct bt_has_preset_record record;
+	struct {
+		uint8_t index;
+		enum bt_has_properties properties;
+		char name[BT_HAS_PRESET_NAME_MAX + 1]; /* + 1 byte for null-terminator */
+	};
+};
+
+static void handle_read_preset_rsp(struct has_inst *inst, struct net_buf_simple *buf)
+{
+	const struct bt_has_cp_read_preset_rsp *pdu;
+	union preset_record record;
+	size_t name_len;
+
+	BT_DBG("conn %p buf %p", (void *)inst->conn, buf);
+
+	if (buf->len < sizeof(*pdu)) {
+		BT_ERR("malformed PDU");
+		return;
+	}
+
+	pdu = net_buf_simple_pull_mem(buf, sizeof(*pdu));
+
+	if (pdu->is_last > 0x01) {
+		BT_WARN("unexpected is_last value 0x%02x", pdu->is_last);
+	}
```

Comment:
Please use a `#define` instead of magic numbers for the `is_last` value check :)